hbase-client/src/main/java/org/apache/hadoop/hbase/client/HRegionLocator.java

```diff
@@ -122,6 +135,37 @@ public boolean visitInternal(Result result) throws IOException {
       }
     };
     MetaTableAccessor.scanMetaForTableRegions(connection, visitor, tableName);
-    return regions;
+    return consolidate(regions);
```

Comment:
Same as before, different regions cannot be consolidated into one HRegionLocation array as their start_keys are different.

Reply:
Test failures such as https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4014/1/testReport/org.apache.hadoop.hbase.client/TestAdmin/precommit_checks___yetus_jdk11_hadoop3_checks___testCreateTableWithRegions is probably caused by this change as 10 regions is only 1 now.

Reply:
I think you are correct. I see now that `RegionLocations` has a class-level javadoc, which explains some of this. I wish these methods described their return values.

I have reverted these changes and `TestAdmin` passes locally. Let me push that commit and see if it resolves all of the test failures.

Reply:
The precommit check passes/